language-support/hs/bindings/examples/group-chat/daml/GroupChat.daml

```diff
@@ -79,113 +81,125 @@ defaulting x y = Some (DA.Optional.fromOptional y x)
 
 -- Testing
 
-only_members_see_messages = scenario do
-
-    alice <- getParty "Alice"
-    bob   <- getParty "Bob"
-    chris <- getParty "Chris"
-
-    g  <- submit alice do create     Group        with members = [alice], invitees = [], origin = None
-    g  <- submit alice do exercise g Group_Invite with inviter = alice, candidate = bob
-    g  <- submit alice do exercise g Group_Invite with inviter = alice, candidate = chris
-    g  <- submit bob   do exercise g Group_Join   with joiner = bob
-    m1 <- submit bob   do exercise g Group_Send   with sender = bob, body = "From Alice"
-    g  <- submit chris do exercise g Group_Join   with joiner = chris
-    m2 <- submit bob   do exercise g Group_Send   with sender = bob, body = "From Bob"
-    g  <- submit alice do exercise g Group_Leave  with leaver = alice
-    m3 <- submit chris do exercise g Group_Send   with sender = chris, body = "From Chris"
-
-    submit         alice do fetch m1
-    submit         bob   do fetch m1
-    submitMustFail chris do fetch m1
-
-    submit         alice do fetch m2
-    submit         bob   do fetch m2
-    submit         chris do fetch m2
-
-    submitMustFail alice do fetch m3
-    submit         bob   do fetch m3
-    submit         chris do fetch m3
+template Helper
+  with
+    p : Party
+  where
+    signatory p
+
+    choice Fetch : Message
```

Comment:
is this something you could have done with `queryContractId`?

Reply:
Good point! yes I could have done that in this case, I think I just went with the `Helper` out of habit after adapting another test which did need it. I can rewrite lines 112-122 to use `queryContractId` instead if you want

Reply:
not an important change, I was asking more for my understanding.

Reply:
fair :)